### PR TITLE
Implement conversion webhooks

### PR DESCRIPTION
**Pull request by fedepaol**
Based off https://github.com/metallb/metallb/pull/1291
We add conversion webhooks for addresspool v1alpha1, and for bgpeer v1beta1 to v1beta2.
Plus, we add a new CI lane that runs a selected pool of tests from v0.12.1 in operator mode, to ensure that the old api still works.

### CODE REVIEW

File: api/v1beta1/bgppeer_conversion.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		return err
+	}
+	return nil
+}
+
+func parseNodeSelectors(selectors []NodeSelector) ([]metav1.LabelSelector, error) {
+	var nodeSels []metav1.LabelSelector
+	if len(selectors) == 0 {
+		return []metav1.LabelSelector{}, nil
+	}
```
Comment (oribon) [inline api/v1beta1/bgppeer_conversion.go:63]: nit: can we drop this?

File: api/v1beta1/bgppeer_conversion.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		sel.MatchLabels[k] = v
+	}
+	for _, req := range ns.MatchExpressions {
+		sel.MatchExpressions = append(sel.MatchExpressions, metav1.LabelSelectorRequirement{
+			Key:      req.Key,
+			Operator: metav1.LabelSelectorOperator(req.Operator),
+			Values:   req.Values,
+		})
+	}
+	return metav1.LabelSelector{}, nil
```
Comment (oribon) [inline api/v1beta1/bgppeer_conversion.go:92]: shouldn't this be `sel, nil` ?

Comment (fedepaol) [reply api/v1beta1/bgppeer_conversion.go:92]: indeed

File: api/v1beta1/bgppeer_conversion.go (hunk 1 above)
Comment (fedepaol) [reply api/v1beta1/bgppeer_conversion.go:63]: yep

File: api/v1beta1/bgppeer_conversion.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	var err error
+	dst.Spec.NodeSelectors, err = parseNodeSelectors(src.Spec.NodeSelectors)
+	if err != nil {
+		return err
+	}
+	dst.ObjectMeta = src.ObjectMeta
+	return nil
+}
+
+// ConvertFrom converts from the Hub version (vbeta2) to this version.
```
Comment (sabinaaledort) [inline api/v1beta1/bgppeer_conversion.go:36]: vbeta2 -> v1beta2

Comment (fedepaol) [reply api/v1beta1/bgppeer_conversion.go]: fixed

**Comment by oribon** [comment]
LGTM

**Pull request merged**